model/riscv_pmp_control.sail

```diff
@@ -152,104 +158,39 @@ function pmpMatchEntry(addr: xlenbits, width: xlenbits, acc: AccessType(ext_acce
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp2cfg, pmpaddr2, pmpaddr1) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp3cfg, pmpaddr3, pmpaddr2) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp4cfg, pmpaddr4, pmpaddr3) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp5cfg, pmpaddr5, pmpaddr4) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp6cfg, pmpaddr6, pmpaddr5) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp7cfg, pmpaddr7, pmpaddr6) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp8cfg, pmpaddr8, pmpaddr7) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp9cfg, pmpaddr9, pmpaddr8) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp10cfg, pmpaddr10, pmpaddr9) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp11cfg, pmpaddr11, pmpaddr10) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp12cfg, pmpaddr12, pmpaddr11) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp13cfg, pmpaddr13, pmpaddr12) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp14cfg, pmpaddr14, pmpaddr13) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue =>
-  match pmpMatchEntry(addr, width, acc, priv, pmp15cfg, pmpaddr15, pmpaddr14) {
-    PMP_Success  => true,
-    PMP_Fail     => false,
-    PMP_Continue => match priv {
-                      Machine => true,
-                      _       => false
-                    }
-  }}}}}}}}}}}}}}}};
-
-  if   check
-  then None()
-  else match acc {
-    Read(_)      => Some(E_Load_Access_Fault()),
-    Write(_)     => Some(E_SAMO_Access_Fault()),
-    ReadWrite(_) => Some(E_SAMO_Access_Fault()),
-    Execute()    => Some(E_Fetch_Access_Fault())
-  }
+
+  foreach (i from 0 to 15) {
+    let prev_pmpaddr = (if i > 0 then pmpReadAddrReg(i) else zeros());
+    match pmpMatchEntry(addr, width, acc, priv, pmpcfg_n[i], pmpReadAddrReg(i), prev_pmpaddr) {
+      PMP_Success  => { return None(); },
+      PMP_Fail     => { return Some(accessToFault(acc)); },
+      PMP_Continue => (),
+    }
+  };
+  if priv == Machine then None() else Some(accessToFault(acc))
```

Comment:
I am not a Sail expert, but this appears to let Mmode to never trap, when in fact if the Lock bit is set in config, it has to pass the same checks as if  it were not in Mmode

Reply:
OK. I wish there were an emacs-like tag facility in github so I could track those dependencies easily.

Reply:
I agree! It does actually support that, but only for some languages and apparently not in PRs. If you use VSCode you can open PRs in VSCode, but it's not that great tbh.